### ISSUE TITLE
Deployment on Heroku

### ISSUE THREAD

**Issue by dedman** [palkan/logidze#11]
First of all, awesome gem! 

I just tried to deploy to Heroku, and the migration returned this error.

`ActiveRecord::StatementInvalid: PG::InsufficientPrivilege: ERROR:  permission denied to set parameter "logidze.disabled"
CONTEXT:  SQL statement "ALTER DATABASE ***DB_NAME*** SET logidze.disabled TO off"`

Any ideas on how this might be fixed?

**Comment by palkan** [comment]
Logidze uses custom database variables (to conditionally track changes and responsible users) and thus requires superuser privileges (unless 9.6 haven't been released).

And Heroku doesn't provide superuser access (or other ways to alter databases or roles).

So, I had no idea how we can fix this for 9.5 so far. Sorry( 
I need a little bit more time for investigation.

**Comment by nobodyzzz** [comment]
Face the same issue on Amazon RDS, as workaround use this:
```ruby
require 'active_record/connection_adapters/postgresql_adapter'

ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.set_callback :checkout, :after do
  execute("SELECT set_config('logidze.disabled', 'off', false)")
  execute("SELECT set_config('logidze.responsible', 'off', false)")
end
```
(Put to config/initializers/logidze_workaround.rb)

**Comment by palkan** [comment]
@nobodyzzz Thanks for the workaround! Added to the readme.

I just want to add that you still have to remove "ALTER DATABASE ..." statements from the installation migration.

**Comment by palkan** [comment]
@dedman 
I've updated the gem (version 0.3.1), now should work on Heroku with PostgreSQL 9.6 without any tweak.

**Comment by dedman** [comment]
Thanks @palkan!

**Comment by Frexuz** [comment]
Then why is the alter database statements still in the generated migration template if it's not needed? 🤔

**Comment by palkan** [comment]
These statements are only included for PostgreSQL <9.6 where we need them (see https://github.com/palkan/logidze/issues/11#issuecomment-238701375)

**Comment by Frexuz** [comment]
Ah, my mistake :) Thanks
